src/core/analysis-context.ts

```diff
@@ -247,16 +247,26 @@ export class AnalysisContext {
   }
 
   /**
-   * Return a copy, but with the given cache.
+   * Returns a copy of the context but with optional replacements of cache or
+   * constructor options.
+   *
+   * Note: this feature is experimental.
    */
-  private _fork(cache: AnalysisCache): AnalysisContext {
-    const copy = new AnalysisContext({
+  _fork(cache?: AnalysisCache, options?: ForkOptions): AnalysisContext {
+    const contextOptions: Options = {
       lazyEdges: this._lazyEdges,
       parsers: this._parsers,
       scanners: this._scanners,
       urlLoader: this._loader,
-      urlResolver: this._resolver
-    });
+      urlResolver: this._resolver,
+    };
+    if (options) {
+      Object.assign(contextOptions, options);
+    }
+    const copy = new AnalysisContext(contextOptions);
+    if (!cache) {
+      cache = this._cache.invalidate([]);
+    }
     copy._cache = cache;
```

Comment:
That's fair. Please also update the code for _fork to not use Object.assign but to only override the one known property.

Also please document on ForkOptions.urlLoader that forking does not clear caches, so you will see a mixture of pre-fork and post-fork contents when you analyze with a forked analyzer.